### PR TITLE
feat: migrate editor-ui to Vite.js and various DX improvements (N8N-2277)

### PR DESCRIPTION
**Pull request by alexgrozav**

### CODE REVIEW

File: packages/editor-ui/src/n8n-theme.scss [hunk 1]
```diff
@@ -1,7 +1,5 @@
 @import "./n8n-theme-variables";
 
-@import "~n8n-design-system/theme/dist/index.css";
```
Comment (mutdmour) [inline packages/editor-ui/src/n8n-theme.scss:3]: how are we importing this now?

**Review by mutdmour — COMMENTED** [review]
Looks good overall. Cannot approve because of outstanding issues that need to be resolved

File: packages/workflow/src/Interfaces.ts [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
 import * as express from 'express';
 import * as FormData from 'form-data';
 import { URLSearchParams } from 'url';
 import { IDeferredPromise } from './DeferredPromise';
-import { Workflow } from './Workflow';
-import { WorkflowHooks } from './WorkflowHooks';
-import { WorkflowActivationError } from './WorkflowActivationError';
-import { WorkflowOperationError } from './WorkflowErrors';
-import { NodeApiError, NodeOperationError } from './NodeErrors';
+import type { Workflow } from './Workflow';
```
[the remark this answers is not in the document]
Comment (alexgrozav) [reply packages/workflow/src/Interfaces.ts:9]: It's a typescript feature that allows you to import declarations only, comes in handy when you need to remove circular references and imports.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

File: packages/editor-ui/src/n8n-theme.scss (hunk 1 above)
Comment (alexgrozav) [reply packages/editor-ui/src/n8n-theme.scss:3]: We're importing it directly in `design-system/src/main.js`. Doing so allows the build system to split into chunks and write font/asset paths properly.

https://github.com/n8n-io/n8n/pull/4061/files/0ab42166505e390ec27336f4c3e934eb5cac7c45#diff-2396dce48aeead825ea11d2267d6213773ebcb0887535809073d8ea228daed3b

File: packages/workflow/src/Interfaces.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ivov) [reply packages/workflow/src/Interfaces.ts:9]: They should all be type imports.

File: packages/workflow/tsconfig.test.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+	"extends": "./tsconfig.json",
+	"include": [
+		"test/**/*.d.ts",
+		"test/**/*"
+	]
+}
```
Comment (ivov) [inline packages/workflow/tsconfig.test.json:7]: Why is this needed?

File: packages/editor-ui/.npmignore [hunk 4]
```diff
@@ -4,7 +4,6 @@
 
 .browserslistrc
 babel.config.js
-cypress.json
```
Comment (ivov) [inline packages/editor-ui/.npmignore:7]: Why ship this?

File: packages/editor-ui/index.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
```
Comment (ivov) [inline packages/editor-ui/index.html:1]: To learn: This does not belong in `/public` anymore?

File: packages/editor-ui/package.json [hunk 6]
```diff
@@ -14,33 +14,63 @@
...
     "dev": "npm run serve",
-    "format": "cd ../.. && node_modules/prettier/bin-prettier.js packages/editor-ui/**/**.ts --write",
     "lint": "tslint -p tsconfig.json -c tslint.json && eslint .",
     "lintfix": "tslint --fix -p tsconfig.json -c tslint.json && eslint . --fix",
-    "serve": "cross-env VUE_APP_URL_BASE_API=http://localhost:5678/ vue-cli-service serve",
-    "test:e2e": "vue-cli-service test:e2e",
-    "test:unit": "vue-cli-service test:unit"
+    "format": "cd ../.. && node_modules/prettier/bin-prettier.js packages/editor-ui/**/**.ts --write",
+    "serve": "cross-env VUE_APP_URL_BASE_API=http://localhost:5678/ vite --host 0.0.0.0 --port 8080 dev",
+    "test": "# vitest run",
```
Comment (ivov) [inline packages/editor-ui/package.json:23]: `#`?

**Review by ivov — COMMENTED** [review]
One lingering concern. On refreshing or returning to the canvas, `master` is perceptively faster than the Vite branch on 5678 and 8080, although metrics like `DOMContentLoaded` are much better.

**Comment by ivov** [comment]
1. Getting this error on 5678 when switching locale:

```
Uncaught (in promise) Error: Unknown variable dynamic import: ./locales/de.json
    __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
    __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
    loadLanguage http://localhost:5678/assets/index.754241ce.js:16669
    defaultLocale http://localhost:5678/assets/index.754241ce.js:16669
    invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
    run http://localhost:5678/assets/index.754241ce.js:5
    flushSchedulerQueue http://localhost:5678/assets/index.754241ce.js:5
    nextTick http://localhost:5678/assets/index.754241ce.js:5
    flushCallbacks http://localhost:5678/assets/index.754241ce.js:5
    promise callback*timerFunc http://localhost:5678/assets/index.754241ce.js:5
    nextTick http://localhost:5678/assets/index.754241ce.js:5
    queueWatcher http://localhost:5678/assets/index.754241ce.js:5
    update http://localhost:5678/assets/index.754241ce.js:5
    notify http://localhost:5678/assets/index.754241ce.js:5
    set http://localhost:5678/assets/index.754241ce.js:5
    set$1 http://localhost:5678/assets/index.754241ce.js:5
    setDefaultLocale http://localhost:5678/assets/index.754241ce.js:16874
    registerMutation http://localhost:5678/assets/index.754241ce.js:609
    commit http://localhost:5678/assets/index.754241ce.js:609
    commit http://localhost:5678/assets/index.754241ce.js:609
    _withCommit http://localhost:5678/assets/index.754241ce.js:609
    commit http://localhost:5678/assets/index.754241ce.js:609
    commit http://localhost:5678/assets/index.754241ce.js:609
    commit http://localhost:5678/assets/index.754241ce.js:609
    getSettings http://localhost:5678/assets/index.754241ce.js:16874
    registerAction http://localhost:5678/assets/index.754241ce.js:609
    dispatch http://localhost:5678/assets/index.754241ce.js:609
    dispatch http://localhost:5678/assets/index.754241ce.js:609
    initSettings http://localhost:5678/assets/index.754241ce.js:16669
    initialize http://localhost:5678/assets/index.754241ce.js:16669
    mounted http://localhost:5678/assets/index.754241ce.js:16669
    invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
    callHook http://localhost:5678/assets/index.754241ce.js:5
    insert http://localhost:5678/assets/index.754241ce.js:5
    z http://localhost:5678/assets/index.754241ce.js:5
    createPatchFunction http://localhost:5678/assets/index.754241ce.js:5
    _update http://localhost:5678/assets/index.754241ce.js:5
    u http://localhost:5678/assets/index.754241ce.js:5
    get http://localhost:5678/assets/index.754241ce.js:5
    Watcher http://localhost:5678/assets/index.754241ce.js:5
    mountComponent http://localhost:5678/assets/index.754241ce.js:5
    $mount http://localhost:5678/assets/index.754241ce.js:5
    <anonymous> http://localhost:5678/assets/index.754241ce.js:16874
```

To reproduce, create `packages/editor-ui/src/plugins/i18n/locales/de.json`, edit all values to make it visually clear it is a different locale, `export N8N_DEFAULT_LOCALE=de` on 5678 and start. Editing the `de.json` triggers hot reload on 8080, but its translated strings are not used.

File: packages/workflow/tsconfig.test.json (hunk 3 above)
Comment (alexgrozav) [reply packages/workflow/tsconfig.test.json:7]: Our `tsconfig.json` only includes `src/**/*.ts`. This will add typescript coverage to our `test` folder without adding it to the build output.

File: packages/editor-ui/index.html (hunk 5 above)
Comment (alexgrozav) [reply packages/editor-ui/index.html:1]: Vite.js expects `index.html` to be in the project root and have references to build entry points. Everything in `public` will still be copied over though.

**Comment by alexgrozav** [comment]
> One lingering concern. On refreshing or returning to the canvas, `master` is perceptively faster than the Vite branch on 5678 and 8080, although metrics like `DOMContentLoaded` are much better.

@ivov Only the first few loads will be slower until Vite creates the dependencies graph and caches it. Subsequent starts are much faster than webpack. At least that's how it is for me.

**Comment by alexgrozav** [comment]
> Found this error on 5678 when switching locale:
> 
> ```
> Uncaught (in promise) Error: Unknown variable dynamic import: ./locales/de.json
>     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
>     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
>     loadLanguage http://localhost:5678/assets/index.754241ce.js:16669
>     defaultLocale http://localhost:5678/assets/index.754241ce.js:16669
>     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
>     run http://localhost:5678/assets/index.754241ce.js:5
>     flushSchedulerQueue http://localhost:5678/assets/index.754241ce.js:5
>     nextTick http://localhost:5678/assets/index.754241ce.js:5
>     flushCallbacks http://localhost:5678/assets/index.754241ce.js:5
>     promise callback*timerFunc http://localhost:5678/assets/index.754241ce.js:5
>     nextTick http://localhost:5678/assets/index.754241ce.js:5
>     queueWatcher http://localhost:5678/assets/index.754241ce.js:5
>     update http://localhost:5678/assets/index.754241ce.js:5
>     notify http://localhost:5678/assets/index.754241ce.js:5
>     set http://localhost:5678/assets/index.754241ce.js:5
>     set$1 http://localhost:5678/assets/index.754241ce.js:5
>     setDefaultLocale http://localhost:5678/assets/index.754241ce.js:16874
>     registerMutation http://localhost:5678/assets/index.754241ce.js:609
>     commit http://localhost:5678/assets/index.754241ce.js:609
>     commit http://localhost:5678/assets/index.754241ce.js:609
>     _withCommit http://localhost:5678/assets/index.754241ce.js:609
>     commit http://localhost:5678/assets/index.754241ce.js:609
>     commit http://localhost:5678/assets/index.754241ce.js:609
>     commit http://localhost:5678/assets/index.754241ce.js:609
>     getSettings http://localhost:5678/assets/index.754241ce.js:16874
>     registerAction http://localhost:5678/assets/index.754241ce.js:609
>     dispatch http://localhost:5678/assets/index.754241ce.js:609
>     dispatch http://localhost:5678/assets/index.754241ce.js:609
>     initSettings http://localhost:5678/assets/index.754241ce.js:16669
>     initialize http://localhost:5678/assets/index.754241ce.js:16669
>     mounted http://localhost:5678/assets/index.754241ce.js:16669
>     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
>     callHook http://localhost:5678/assets/index.754241ce.js:5
>     insert http://localhost:5678/assets/index.754241ce.js:5
>     z http://localhost:5678/assets/index.754241ce.js:5
>     createPatchFunction http://localhost:5678/assets/index.754241ce.js:5
>     _update http://localhost:5678/assets/index.754241ce.js:5
>     u http://localhost:5678/assets/index.754241ce.js:5
>     get http://localhost:5678/assets/index.754241ce.js:5
>     Watcher http://localhost:5678/assets/index.754241ce.js:5
>     mountComponent http://localhost:5678/assets/index.754241ce.js:5
>     $mount http://localhost:5678/assets/index.754241ce.js:5
>     <anonymous> http://localhost:5678/assets/index.754241ce.js:16874
> ```
> 
> To reproduce, create `packages/editor-ui/src/plugins/i18n/locales/de.json`, edit all values to make it visually clear it is a different locale, `export N8N_DEFAULT_LOCALE=de` on 5678 and start. Editing the `de.json` triggers hot reload on 8080, but its translated strings are not used. [Docs link](https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/plugins/i18n/docs/README.md#building-translations).

Good catch, working on fixing it!

**Comment by ivov** [comment]
Whitespace and ampersands missing on node panel category headers:

image

File: packages/editor-ui/package.json (hunk 6 above)
Comment (alexgrozav) [reply packages/editor-ui/package.json:23]: > > Found this error on 5678 when switching locale:
> > ```
> > Uncaught (in promise) Error: Unknown variable dynamic import: ./locales/de.json
> >     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
> >     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
> >     loadLanguage http://localhost:5678/assets/index.754241ce.js:16669
> >     defaultLocale http://localhost:5678/assets/index.754241ce.js:16669
> >     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
> >     run http://localhost:5678/assets/index.754241ce.js:5
> >     flushSchedulerQueue http://localhost:5678/assets/index.754241ce.js:5
> >     nextTick http://localhost:5678/assets/index.754241ce.js:5
> >     flushCallbacks http://localhost:5678/assets/index.754241ce.js:5
> >     promise callback*timerFunc http://localhost:5678/assets/index.754241ce.js:5
> >     nextTick http://localhost:5678/assets/index.754241ce.js:5
> >     queueWatcher http://localhost:5678/assets/index.754241ce.js:5
> >     update http://localhost:5678/assets/index.754241ce.js:5
> >     notify http://localhost:5678/assets/index.754241ce.js:5
> >     set http://localhost:5678/assets/index.754241ce.js:5
> >     set$1 http://localhost:5678/assets/index.754241ce.js:5
> >     setDefaultLocale http://localhost:5678/assets/index.754241ce.js:16874
> >     registerMutation http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     _withCommit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     getSettings http://localhost:5678/assets/index.754241ce.js:16874
> >     registerAction http://localhost:5678/assets/index.754241ce.js:609
> >     dispatch http://localhost:5678/assets/index.754241ce.js:609
> >     dispatch http://localhost:5678/assets/index.754241ce.js:609
> >     initSettings http://localhost:5678/assets/index.754241ce.js:16669
> >     initialize http://localhost:5678/assets/index.754241ce.js:16669
> >     mounted http://localhost:5678/assets/index.754241ce.js:16669
> >     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
> >     callHook http://localhost:5678/assets/index.754241ce.js:5
> >     insert http://localhost:5678/assets/index.754241ce.js:5
> >     z http://localhost:5678/assets/index.754241ce.js:5
> >     createPatchFunction http://localhost:5678/assets/index.754241ce.js:5
> >     _update http://localhost:5678/assets/index.754241ce.js:5
> >     u http://localhost:5678/assets/index.754241ce.js:5
> >     get http://localhost:5678/assets/index.754241ce.js:5
> >     Watcher http://localhost:5678/assets/index.754241ce.js:5
> >     mountComponent http://localhost:5678/assets/index.754241ce.js:5
> >     $mount http://localhost:5678/assets/index.754241ce.js:5
> >     <anonymous> http://localhost:5678/assets/index.754241ce.js:16874
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To reproduce, create `packages/editor-ui/src/plugins/i18n/locales/de.json`, edit all values to make it visually clear it is a different locale, `export N8N_DEFAULT_LOCALE=de` on 5678 and start. Editing the `de.json` triggers hot reload on 8080, but its translated strings are not used. [Docs link](https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/plugins/i18n/docs/README.md#building-translations).
> 
> Good catch, working on fixing it!

Looks like we both contributed a bit to this 😅. Fixed!

**Comment by alexgrozav** [comment]
> > Found this error on 5678 when switching locale:
> > ```
> > Uncaught (in promise) Error: Unknown variable dynamic import: ./locales/de.json
> >     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
> >     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
> >     loadLanguage http://localhost:5678/assets/index.754241ce.js:16669
> >     defaultLocale http://localhost:5678/assets/index.754241ce.js:16669
> >     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
> >     run http://localhost:5678/assets/index.754241ce.js:5
> >     flushSchedulerQueue http://localhost:5678/assets/index.754241ce.js:5
> >     nextTick http://localhost:5678/assets/index.754241ce.js:5
> >     flushCallbacks http://localhost:5678/assets/index.754241ce.js:5
> >     promise callback*timerFunc http://localhost:5678/assets/index.754241ce.js:5
> >     nextTick http://localhost:5678/assets/index.754241ce.js:5
> >     queueWatcher http://localhost:5678/assets/index.754241ce.js:5
> >     update http://localhost:5678/assets/index.754241ce.js:5
> >     notify http://localhost:5678/assets/index.754241ce.js:5
> >     set http://localhost:5678/assets/index.754241ce.js:5
> >     set$1 http://localhost:5678/assets/index.754241ce.js:5
> >     setDefaultLocale http://localhost:5678/assets/index.754241ce.js:16874
> >     registerMutation http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     _withCommit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     getSettings http://localhost:5678/assets/index.754241ce.js:16874
> >     registerAction http://localhost:5678/assets/index.754241ce.js:609
> >     dispatch http://localhost:5678/assets/index.754241ce.js:609
> >     dispatch http://localhost:5678/assets/index.754241ce.js:609
> >     initSettings http://localhost:5678/assets/index.754241ce.js:16669
> >     initialize http://localhost:5678/assets/index.754241ce.js:16669
> >     mounted http://localhost:5678/assets/index.754241ce.js:16669
> >     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
> >     callHook http://localhost:5678/assets/index.754241ce.js:5
> >     insert http://localhost:5678/assets/index.754241ce.js:5
> >     z http://localhost:5678/assets/index.754241ce.js:5
> >     createPatchFunction http://localhost:5678/assets/index.754241ce.js:5
> >     _update http://localhost:5678/assets/index.754241ce.js:5
> >     u http://localhost:5678/assets/index.754241ce.js:5
> >     get http://localhost:5678/assets/index.754241ce.js:5
> >     Watcher http://localhost:5678/assets/index.754241ce.js:5
> >     mountComponent http://localhost:5678/assets/index.754241ce.js:5
> >     $mount http://localhost:5678/assets/index.754241ce.js:5
> >     <anonymous> http://localhost:5678/assets/index.754241ce.js:16874
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To reproduce, create `packages/editor-ui/src/plugins/i18n/locales/de.json`, edit all values to make it visually clear it is a different locale, `export N8N_DEFAULT_LOCALE=de` on 5678 and start. Editing the `de.json` triggers hot reload on 8080, but its translated strings are not used. [Docs link](https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/plugins/i18n/docs/README.md#building-translations).
> 
> Good catch, working on fixing it!

> > Found this error on 5678 when switching locale:
> > ```
> > Uncaught (in promise) Error: Unknown variable dynamic import: ./locales/de.json
> >     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
> >     __variableDynamicImportRuntime0__ http://localhost:5678/assets/index.754241ce.js:16669
> >     loadLanguage http://localhost:5678/assets/index.754241ce.js:16669
> >     defaultLocale http://localhost:5678/assets/index.754241ce.js:16669
> >     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
> >     run http://localhost:5678/assets/index.754241ce.js:5
> >     flushSchedulerQueue http://localhost:5678/assets/index.754241ce.js:5
> >     nextTick http://localhost:5678/assets/index.754241ce.js:5
> >     flushCallbacks http://localhost:5678/assets/index.754241ce.js:5
> >     promise callback*timerFunc http://localhost:5678/assets/index.754241ce.js:5
> >     nextTick http://localhost:5678/assets/index.754241ce.js:5
> >     queueWatcher http://localhost:5678/assets/index.754241ce.js:5
> >     update http://localhost:5678/assets/index.754241ce.js:5
> >     notify http://localhost:5678/assets/index.754241ce.js:5
> >     set http://localhost:5678/assets/index.754241ce.js:5
> >     set$1 http://localhost:5678/assets/index.754241ce.js:5
> >     setDefaultLocale http://localhost:5678/assets/index.754241ce.js:16874
> >     registerMutation http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     _withCommit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     commit http://localhost:5678/assets/index.754241ce.js:609
> >     getSettings http://localhost:5678/assets/index.754241ce.js:16874
> >     registerAction http://localhost:5678/assets/index.754241ce.js:609
> >     dispatch http://localhost:5678/assets/index.754241ce.js:609
> >     dispatch http://localhost:5678/assets/index.754241ce.js:609
> >     initSettings http://localhost:5678/assets/index.754241ce.js:16669
> >     initialize http://localhost:5678/assets/index.754241ce.js:16669
> >     mounted http://localhost:5678/assets/index.754241ce.js:16669
> >     invokeWithErrorHandling http://localhost:5678/assets/index.754241ce.js:5
> >     callHook http://localhost:5678/assets/index.754241ce.js:5
> >     insert http://localhost:5678/assets/index.754241ce.js:5
> >     z http://localhost:5678/assets/index.754241ce.js:5
> >     createPatchFunction http://localhost:5678/assets/index.754241ce.js:5
> >     _update http://localhost:5678/assets/index.754241ce.js:5
> >     u http://localhost:5678/assets/index.754241ce.js:5
> >     get http://localhost:5678/assets/index.754241ce.js:5
> >     Watcher http://localhost:5678/assets/index.754241ce.js:5
> >     mountComponent http://localhost:5678/assets/index.754241ce.js:5
> >     $mount http://localhost:5678/assets/index.754241ce.js:5
> >     <anonymous> http://localhost:5678/assets/index.754241ce.js:16874
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To reproduce, create `packages/editor-ui/src/plugins/i18n/locales/de.json`, edit all values to make it visually clear it is a different locale, `export N8N_DEFAULT_LOCALE=de` on 5678 and start. Editing the `de.json` triggers hot reload on 8080, but its translated strings are not used. [Docs link](https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/plugins/i18n/docs/README.md#building-translations).
> 
> Good catch, working on fixing it!

Looks like we both contributed a bit to this 😅. Fixed!

**Comment by alexgrozav** [comment]
> Whitespace and ampersands missing on node panel category headers:
> 
> image

This is translations related. Got fixed together with the previous fix.

File: packages/cli/src/Server.ts [hunk 7]
```diff
@@ -1753,11 +1755,28 @@ class App {
...
 			const filePath = pathJoin(pathDirname(editorUiPath), 'dist', 'index.html');
 			const n8nPath = config.getEnv('path');
+			const basePathRegEx = /\/%BASE_PATH%\//g;
 
 			let readIndexFile = readFileSync(filePath, 'utf8');
-			readIndexFile = readIndexFile.replace(/\/%BASE_PATH%\//g, n8nPath);
+			readIndexFile = readIndexFile.replace(basePathRegEx, n8nPath);
 			readIndexFile = readIndexFile.replace(/\/favicon.ico/g, `${n8nPath}favicon.ico`);
 
+			const cssPath = pathJoin(pathDirname(editorUiPath), 'dist', '**/*.css');
```
Comment (netroy) [inline packages/cli/src/Server.ts:1764]: This change means that we'll be keeping a copy of all JS and CSS files in the memory, and increasing the overall memory usage of every instance. Also, this makes gzipping these resources more difficult, which is not really an issue as we are currently not compressing any of our assets (from what I can tell).
However. this isn't a blocker. I can take care of optimizing these and setting up compression in another PR.

Comment (alexgrozav) [reply packages/cli/src/Server.ts:1764]: Thanks a lot, Adi. I've crated a task for it: https://linear.app/n8n/issue/N8N-4739

**Comment by netroy** [comment]
> > One lingering concern. On refreshing or returning to the canvas, `master` is perceptively faster than the Vite branch on 5678 and 8080, although metrics like `DOMContentLoaded` are much better.
> 
> @ivov Only the first few loads will be slower until Vite creates the dependencies graph and caches it. Subsequent starts are much faster than webpack. At least that's how it is for me.

I'm seeing this issue on my staging instance. every reload takes significantly longer than it used to. even though none of the http requests seem slow.

**Comment by alexgrozav** [comment]
Interesting. I see now. I thought @ivov was referring to initial page load. I'll try to do something about this.

File: packages/editor-ui/src/modules/nodeTypes.ts [hunk 8]
```diff
@@ -58,8 +58,7 @@ const module: Module<INodeTypesState, IRootState> = {
 	},
 	mutations: {
 		setNodeTypes(state, newNodeTypes: INodeTypeDescription[]) {
-			// tslint:disable-next-line:no-any
-			const nodeTypes = newNodeTypes.reduce((acc: any, newNodeType) => {
+			const nodeTypes = newNodeTypes.reduce((acc: Record<string, Record<string, INodeTypeDescription>>, newNodeType) => {
```
Comment (ivov) [inline packages/editor-ui/src/modules/nodeTypes.ts:61]: Nit: The type for the accumulator is provided [as a generic](https://github.com/microsoft/TypeScript/blob/afee8bf9682dfb3ed927fdc7756d96a19851b811/lib/lib.es5.d.ts#L1386) to `reduce`

Comment (alexgrozav) [reply packages/editor-ui/src/modules/nodeTypes.ts]: Today I learned 😃

**Review by mutdmour — APPROVED** [review]
🍻

**Comment by janober** [comment]
Got released with `n8n@0.196.0`
